### PR TITLE
WIP Add animation to numpad and button component

### PR DESCRIPTION
**Pull request by Ferossgp**

### CODE REVIEW

**Comment by errorists** [comment]
comment on first round of tests with a couple buttons that already work
animation appears smooth if the UI thread isn't clogged (see invite friends button), I believe the easing curve, properties and timing are also spot on, tho it might require more testing. The light knock haptic feedback effect we have on iOS is super subtle and very nice. On Android I didn't notice any, which is good as my device doesn't seem to have a haptic feedback engine. There's the added ripple effect that Android had for years now, I think it's a nice touch, we should keep it.

👏 great 👏  work! 👏

File: src/status_im/network/ui/edit_network/views.cljs [hunk 1]
```diff
@@ -62,8 +62,8 @@
               :on-change-text #(re-frame/dispatch [::network/input-changed :network-id %])}])]]
         [react/view styles/bottom-container
          [react/view components.styles/flex]
-         [components.common/bottom-button
-          {:forward?  true
+         [button/button
+          {:type      :next
            :label     (i18n/label :t/save)
```
Comment (Ferossgp) [inline src/status_im/network/ui/edit_network/views.cljs]: remove i18n/label on all migrated buttons

File: src/status_im/ui/components/button.cljs [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
             [status-im.utils.label :as utils.label]))
 
 (defn style-container [type disabled? theme]
   (merge
-   (when (= type :main)
-     {:margin-vertical 8 :margin-horizontal 16})
+   (when (= type :main))
    (cond
      (#{:main :secondary} type)
      {:padding-horizontal 32}
```
Comment (Ferossgp) [inline src/status_im/ui/components/button.cljs:13]: Review sizes, some places adds too much outside spacing.

File: src/status_im/ui/components/button/animated.cljs [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+
+(defn aaa []
+  (prn ":aaasdads"))
+
+(defn button-press-animation
+  [{:keys [animation-state duration-val finished frame-time gesture-state
+           prev-gesture-state scale-value time to-value zoom-clock scale-to
+           on-press on-press-start on-long-press create-interaction reset]}]
+  ;; TODO: Check that animation is not running before call of reset
+  (reanimated/block [(reanimated/cond* (reanimated/or* (reanimated/eq gesture-state (:cancelled reanimated/states))
```
Comment (Ferossgp) [inline src/status_im/ui/components/button/animated.cljs]: When cancelled or failed do not call press handler.

**Comment by errorists** [comment]
Issue 1:
Can't cancel taps. Once you press your thumb on any button, there's no way going back :( The right way: you can remove your finger from the button and then lift it, cancelling the press.

Issue 2:
Some buttons are missing text labels and using incorrect styling, for example here the margin surrounding the button is insane

IMG_1359


Issue 3: 
Floating buttons aren't using this touchable style (Chat +, Wallet Send, Browser select account)

Issue 4:
When I drag on bottom sheets, it registers button presses, so you can't drag the sheet without accidentally pressing any of the buttons :(

**Comment by Ferossgp** [comment]
Animations will be picked when the Button Component will be added into Quo. Now closing as the PR is outdated

**Pull request closed without merging**